omnigibson/envs/env_base.py

```diff
@@ -502,67 +581,52 @@ def step(self, action):
                 of actions
 
         Returns:
-            4-tuple:
+            5-tuple:
                 - dict: state, i.e. next observation
                 - float: reward, i.e. reward at this current timestep
-                - bool: done, i.e. whether this episode is terminated
+                - bool: terminated, i.e. whether this episode ended due to a failure or success
+                - bool: truncated, i.e. whether this episode ended due to a time limit etc.
                 - dict: info, i.e. dictionary with any useful information
         """
         try:
-            # If the action is not a dictionary, convert into a dictionary
-            if not isinstance(action, dict) and not isinstance(action, gym.spaces.Dict):
-                action_dict = dict()
-                idx = 0
-                for robot in self.robots:
-                    action_dim = robot.action_dim
-                    action_dict[robot.name] = action[idx : idx + action_dim]
-                    idx += action_dim
-            else:
-                # Our inputted action is the action dictionary
-                action_dict = action
-
-            # Iterate over all robots and apply actions
-            for robot in self.robots:
-                robot.apply_action(action_dict[robot.name])
-
-            # Run simulation step
+            self._pre_step(action)
             og.sim.step()
-
-            # Grab observations
-            obs, obs_info = self.get_obs()
-
-            # Step the scene graph builder if necessary
-            if self._scene_graph_builder is not None:
-                self._scene_graph_builder.step(self.scene)
-
-            # Grab reward, done, and info, and populate with internal info
-            reward, done, info = self.task.step(self, action)
-            self._populate_info(info)
-            info["obs_info"] = obs_info
-
-            if done and self._automatic_reset:
-                # Add lost observation to our information dict, and reset
-                info["last_observation"] = obs
-                obs = self.reset()
-
-            # Increment step
-            self._current_step += 1
-
-            return obs, reward, done, info
+            return self._post_step(action)
```

Comment:
I don't think it's returning the right thing. Is the new gym API requiring us to return `terminated` and `truncated` rather than `done`?

Reply:
I think `_post_step` here returns `obs, reward, done, info`

Reply:
Yup. @hang-yin please fix